app/grants/clr.py

```diff
@@ -181,27 +182,60 @@ def get_totals_by_pair(contrib_dict):
 
         returns:
             pair totals between current round
-                {user_id (str): {user_id (str): pair_total (float)}}
-
+                {
+                    user_id (str): {
+                        user_id (str): pair_total (float)
+                    }
+                }
+            curr_agg_sqrts
+                {
+                    grantId (int): {
+                        user_id (float): sqrt(aggregated_amount) (float)
+                    }
+                }
+            pairs_tot
+                int: the number of cycles it took to construct the pairs
     '''
+    pairs_tot = 0
     pair_totals = {}
-
+    curr_agg_sqrts = {}
     # start pairwise match
-    for _, contribz in contrib_dict.items():
+    for proj, contribz in curr_agg.items():
+        # build inner layer against contributors
+        contributors = []
+        # cache sqrts of each contribution to make pairing easier
+        curr_agg_sqrts[proj] = {}
+        # iterate on the contributions and set default state
         for k1, v1 in contribz.items():
-            if k1 not in pair_totals:
-                pair_totals[k1] = {}
+            # cache the sqrt of the given value (v1 = total contribution to this grant by this user)
+            curr_agg_sqrts[proj][k1] = float(v1) ** 0.5
```

Comment:
what if we calculate the sqrt when we save the `Contribution` and then we dont have to calculate it again every time we run clr calcs?

this assumes that the sqrt part is costly performance wise at all ( i dont know that it is )

Reply:
oh wait that wont work bc u actually want the` sqrt (all of a users contributions)` not the `sum(sqrt(contribution) x num_contribs)`